docs/src/contributing.md

```diff
@@ -0,0 +1,112 @@
+# Contributing
+
+## Tool Setup
+
+### `python`
+
+To contribute code or documentation updates, an installation of Python 3 is required.
+
+### `hatch`
+
+This project utilizes [`hatch`](https://hatch.pypa.io/latest/) to manage Python environments for development and testing.  Follow
+[the `hatch` installation instructions](https://hatch.pypa.io/latest/install/) before continuing through this document.
+
+### `pre-commit`
+
+Additionally, this project uses [`pre-commit`](https://pre-commit.com/) Git hooks to run linting and formatting checks against each commit.  See [the `pre-commit` installation instructions](https://pre-commit.com/#install) for how to install this tool.
+
+Once installed, run `pre-commit install` to set up the git hook scripts:
+
+``` shell
+$ pre-commit install
+pre-commit installed at .git/hooks/pre-commit
+```
+
+### Git
+
+Clone the repository using:
+
+``` shell
+git clone https://github.com/Archmonger/ServeStatic.git
+cd ServeStatic
+```
+
+All example commands are exected to be run from the `ServeStatic` folder.
+
+## Code Contributions
+
+Ensure you have followed the [tool setup](tools.md) instructions before following the instructions below.
```

Comment:
Need to change the two references to `tools.md` to `#tool-setup` to fix doc tests